frontend/app/services/user-info-service.ts

```diff
@@ -0,0 +1,29 @@
+import { z } from 'zod';
+
+export type UserInfo = {
+  firstName?: string;
+  lastName?: string;
+  phoneNumber?: string;
+};
+
+export async function getUserId() {
+  return '00000000-0000-0000-0000-000000000000';
+}
+
+export async function getUserInfo(userId: string) {
+  const response = await fetch(`https://api.example.com/users/${userId}`);
+  return z.custom<UserInfo>().parse(await response.json());
+}
+
+export async function updateUserInfo(userId: string, userInfo: UserInfo) {
+  const patches = Object.entries(userInfo)
+    .filter(([key, value]) => value !== null && value !== undefined)
+    .map(([key, value]) => ({ op: 'replace', path: `/${key}`, value: value }));
+
+  if (patches.length > 0) {
+    await fetch(`https://api.example.com/users/${userId}`, {
+      method: 'patch',
```

Comment:
```suggestion
      method: 'PATCH',
```

Reply:
> Any string that is a case-insensitive match for one of the methods in [RFC 9110](https://www.rfc-editor.org/rfc/rfc9110#name-overview) will be uppercased automatically.

<https://developer.mozilla.org/en-US/docs/Web/API/fetch#method>

I can change it if you are adamant. :)

Reply:
Change it please.